src/PersonObjects/Player/PlayerObject.ts

```diff
@@ -181,6 +194,11 @@ export class PlayerObject extends Person implements IPlayer {
         delete player.jobs[loadedCompanyName as CompanyName];
       }
     }
+    // Initialize bitNodeOptions
+    if (!player.bitNodeOptions.initialized) {
```

Comment:
Ergh. Having an `initialized` flag is something I would really, *really* like to avoid. The moment you do this, *everything* has to check whether it's been initialized already, and it's just awful.

In this case, it seems wholly unneeded as well, because the entirety of the Player object is in the same state: A bunch of valid-but-not-correct default values until fromJSON loads the save, after which it has the correct values. Since an empty map is valid (and also correct, for the default value of `sourceFiles`), I see no reason for this flag to exist.

Reply:
This check is only useful in `fromJSON`. In `fromJSON`, when we see an empty JSONMap, there are 2 possible cases:
- The player loads an old save, and there is no `bitNodeOptions` in their save data. The empty JSONMap is just a default value, and we need to initialize it with the current `player.sourceFiles`.
- The empty JSONMap is the "real" value. The player intentionally disabled all SFs, so `bitNodeOptions.activeSourceFiles` is empty.

I use `bitNodeOptions.initialized` to determine if we need to initialize `bitNodeOptions.activeSourceFiles`.

Reply:
If it's only useful for old saves, it shouldn't add a new permanent property. One option would be to use savegame conversion code to do the initialization (in SaveObject.ts)

But actually I think it may make sense to rethink the actual structure of the property we're adding to player. Instead of activeSourceFiles, which is an entire clone of the sourceFiles map, it probably makes more sense to just have a sourceFileOverrides (or something like that) which is normally empty except when a player opts to decrease the effective level of their SF. This slims down the player object in the default case (no overrides) and would also mean no initialization required.